src/jdk.jartool/share/classes/sun/tools/jar/Main.java

```diff
@@ -801,7 +801,9 @@ private void expand(File dir, String[] files, Set<String> cpaths, int version)
             if (f.isFile()) {
                 Entry e = new Entry(f, name, false);
                 if (isModuleInfoEntry(name)) {
-                    moduleInfos.putIfAbsent(name, Files.readAllBytes(f.toPath()));
+                    Long lastModified = f.lastModified() == 0 ? null : f.lastModified();
+                    moduleInfos.putIfAbsent(name,
+                            new StreamedModuleInfoEntry(name, Files.readAllBytes(f.toPath()), lastModified));
```

Comment:
I'd prefer to see this split into two two statements as there is just too much going on one the one line.

Reply:
Done. I have updated the PR to split this line into more than 1 statement.